src/vim.h

```diff
@@ -87,6 +87,11 @@
 # endif
 #endif
 
+// C89 does not define SIZE_MAX
+#ifndef SIZE_MAX
+# define SIZE_MAX sizeof(size_t)
```

Comment:
That looks wrong. Note we have a similar define in `xdiff/xmarcros.h` source

https://github.com/vim/vim/blob/9cb865e95bba333fa7b47c39c2729c125b911e36/src/xdiff/xmacros.h#L28-L30